Lesson17/main.js

```diff
@@ -0,0 +1,120 @@
+const slideList = document.getElementById("js-slid-list");
+const slideWrapper = document.getElementById("js-slide-wrapper");
+
+const createElementWithClassName = (type, className) => {
+  const element = document.createElement(type);
+  element.className = className;
+  return element;
+}
+
+const addLoading = (parent) => {
+  const loading = createElementWithClassName("img", "loading");
+  loading.id = "js-loading";
+  loading.src = "./loading-circle.gif";
+  parent.appendChild(loading);
+}
+
+const removeLoading = () => document.getElementById("js-loading").remove();
+
+const addErrorMessage = (error, parent) => {
+  const errorMessage = createElementWithClassName("p", "error-message");
+  errorMessage.textContent = error;
+  parent.appendChild(errorMessage);
+}
+
+const fetchErrorHandling = async (response) => {
+  if (response.ok) {
+    return await response.json();
+  } else {
+    throw new Error("サーバーエラーが発生しました");
+  }
+}
+
+const getJsonOrError = async (url) => {
+  const response = await fetch(url);
+  const json = await fetchErrorHandling(response);
+  return json;
+}
+
+const getSlideImageData = async () => {
+  return new Promise((resolve) => {
+    setTimeout(() => resolve(getJsonOrError("https://myjson.dit.upm.es/api/bins/gzcn")), 3000);
+  })
+}
+
+const tryGetSlideImageData = async () => {
+  addLoading(slideWrapper);
+  try {
+    return await getSlideImageData();
+  } catch (e) {
+    addErrorMessage(e, slideWrapper);
+  } finally {
+    removeLoading();
+  }
+}
+
+const init = async () => {
+  const result = await tryGetSlideImageData();
+  if (result) {
+    createSlideItem(result);
+    createArrowButtons();
+  }
+}
+
+const createSlideItem = (imageData) => {
+  for (let i = 0; i < imageData.length; i++) {
+    const slideItem = createElementWithClassName("li", "slide-item");
+    const slideImage = createElementWithClassName("img", "slide-img");
+    slideImage.src = imageData[i].image;
+    // 初期設定として 最初の要素にis-displayingクラスを付与
+    if (i === 0) slideItem.classList.add("is-displaying");
```

Comment:
IMO

I would implement here like this
`i === 0 && slideItem.classList.add("is-displaying");`

Reply:
Thank you.
I hadn't studied enough around here, so this gave me another opportunity to learn.

8e72588

Reply:
> `i === 0 && slideItem.classList.add("is-displaying");`

I didn't know how to write it like this.
I learned a lot!!